docs/api/variable.md

```diff
@@ -0,0 +1,230 @@
+
+There are three different scopes in which you can define a variable, depending on how much you want it to be shared.
+
+- for common use cases, use higher scopes to reuse the same variable on multiple dashboards
+- for more specific needs, use lower scopes to restrict the variable availability to a specific set of (or even a
+  single) dashboard(s).
+
+### Dashboard level
+
+That's the usual level to define a variable.
+
+```typescript
+interface DashboardSpec {
+  // ... existing dashboard spec ...
+  variables: VariableSpec[];
+}
+```
+
+### Project level
+
+In case you would like to share a variable across different dashboards in the **same** project, you will need to
+create a `Variable`.
+
+```yaml
+  kind: "Variable"
+  metadata:
+    name: <string>
+    project: <string>
+  spec: <variable_spec>
+```
+
+### Global level
+
+When we talk about scope and user permission in a REST API, the easiest way is to associate one permission per endpoint.
+If we want to provide a variable shared by all projects, then it makes sense to have a different object that is
+living outside a project.
+
+That’s why we have another resource called `GlobalVariable`
+
+```yaml
+  kind: "GlobalVariable"
+  metadata:
+    name: <string>
+  spec: <variable_spec>
+```
+
+## Variable specification
+
+We are supporting two different types of variable. `TextVariable` and `ListVariable`
+
+### TextVariable
+
+```yaml
+kind: "TextVariable"
+spec: <text_spec>
+```
+
+#### `<text_spec>`
+
+```yaml
+  # It is a mandatory attribute when you are defining a variable directly in a dashboard.
+  # If you are creating a GlobalVariable or a Variable, you don't have to use this attribute as it is replaced by metadata.name.
+  # This is the unique name of the variable that can be used in another variable or in the different dashboard to use
+  [ name: <string> ]
+
+  [ display: <display_spec> ]
+  value: <string>
+  [ constant: <boolean> | default = false ]
+```
+
+#### Example
+
+```yaml
+kind: "Variable"
+metadata:
+  name: "text"
+  project: "perses"
+spec:
+  kind: "TextVariable"
+  spec:
+    value: "my text"
+```
+
+Or in case you are defining the variable in a dashboard
+
+```yaml
+  variables:
+    - kind: "TextVariable"
+      spec:
+        name: "text"
+        value: "my text"
+```
+
+### ListVariable
+
+```yaml
+kind: "ListVariable"
+spec: <list_spec>
+```
+
+#### `<list_spec>`
+
+```yaml
+  # It is a mandatory attribute when you are defining a variable directly in a dashboard.
+  # If you are creating a GlobalVariable or a Variable, you don't have to use this attribute as it is replaced by metadata.name.
+  # This is the unique name of the variable that can be used in another variable or in the different dashboard to use
+  [ name: <string> ]
+
+  [ display: <display_spec> ]
+
+  # It can be a single value or a list.
+  [ defaultValue: <string> | <array of string> ]
+  [ allowAllValue: <boolean> | default = false ]
+  [ allMultiple: <boolean> | default = false ]
```

Comment:
it would be cool if someone have a good explanation for each field here. It would help to complete the doc :)

Reply:
I'd say

defaultValue: value from the list to be selected by default.
allowAllValue: whether or not to append the "All" value that allows to select all available values at once.
allowMultiple: whether or not to allow multi-selection of values.